editor/src/components/navigator/navigator-item/component-picker-context-menu.tsx

```diff
@@ -485,7 +489,7 @@ function insertComponentPickerItem(
       if (isRenderPropTarget(insertionTarget)) {
         return [
           setProp_UNSAFE(
-            target,
+            targets[0],
```

Comment:
Better to capture the value with `targets.at(0)`, which handles the empty array case correctly. Otherwise this will shove an undefined value all the way down until something breaks because the subscript type is a little naughty.

Reply:
yes, that was my first try (with the "at") - byt this requires a much larger refactor in many places to support undefined (which they shouldn't). Since the original code expected `ElementPath` up in the calling chain (and I changed it to `ElementPath[]`), I think there won't be any empty array there.

<img width="977" alt="image" src="https://github.com/concrete-utopia/utopia/assets/7003853/f9b8d91d-7122-475b-b1f7-69ff32c1c6a4">